File: python/day1.py
calories = []
with open("../inputs/day1.txt", "r") as f:
    line, total_calories = f.readline(), 0
    while line:
        temp_l = line.strip()
        if temp_l.isdigit():
            total_calories += int(temp_l)
        else:
            calories.append(total_calories)
            total_calories = 0
        line = f.readline()

calories.sort()
# part 1
print(calories[-1])
# part 2
print(sum(calories[-3:]))

File: java/src/solutions/Day1.java
package solutions;

import java.io.IOException;
import java.nio.file.Files;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Day1 extends Day implements ISolution {

    private List<Integer> sum_calories;
    private List<String> lines;

    public Day1(String filename) throws IOException {
        super(filename);
        readFiles();
        sum_calories = new ArrayList<>();
        solveCore();
    }

    @Override
    public void readFiles() throws IOException {
        lines = Files.lines(INPUT_FILE).map(line -> line.strip()).toList();
    }

    private void solveCore() throws IOException {
        var total_calories = 0;
        for (String line : lines) {
            if (line.length() > 0 && line.chars().allMatch(Character::isDigit)) {
                total_calories += Integer.parseInt(line);
            } else {
                sum_calories.add(total_calories);
                total_calories = 0;
            }
        }
        Collections.sort(sum_calories);
    }

    @Override
    public void solvePart1() {
        System.out.println(sum_calories.get(sum_calories.size() - 1));
    }

    @Override
    public void solvePart2() {
        System.out.println(sum_calories.stream().skip(sum_calories.size() - 3).reduce(0, Integer::sum));
    }

}
